src/postfixCompletionProvider.ts

```diff
@@ -113,8 +113,9 @@ export class PostfixCompletionProvider implements vsc.CompletionItemProvider {
 
     const codeBeforeTheDot = fullText.slice(0, dotOffset)
 
-    const source = ts.createSourceFile('test.ts', codeBeforeTheDot, ts.ScriptTarget.ESNext, true, ts.ScriptKind.TSX)
-    const fullSource = ts.createSourceFile('test.ts', fullText, ts.ScriptTarget.ESNext, true, ts.ScriptKind.TSX)
+    const scriptKind = document.languageId === 'typescript' ? ts.ScriptKind.TS : ts.ScriptKind.TSX

```

Comment:
This extension is meant to be used in JS as well so `javascript` should parse as `ScriptKind.TS` too.

Reply:
@ipatalas This is intentional. Jsx syntax is fully valid in javascript (that's why next.js demo uses .js ext), but aforementioned generic syntax is valid only in typescript.

Reply:
Maybe you right but with that said I absolutely have no idea what's the difference between JS, JSX and TSX in parsing

Reply:
Yeah, but doesn't it have to be `javascriptreact` in VS Code to fully support it?

Reply:
You can use jsx in .js file with `javascript` language mode (probably thats because js doesn't have generics syntax). But I still decided to go strict with it to avoid bugs in future.

Reply:
All right, I've seen your next commit and this should be ok now except https://github.com/ipatalas/vscode-postfix-ts/pull/100#discussion_r1087164331